### PR TITLE
Filter remotes

### PR DESCRIPTION
**Pull request by ikapelyukhin**
Filtering MPs based on target repo: merging a branch from a completely unrelated repo doesn't make much sense and it's hard to find the needed MP when you have a lot of MPs.

Dropped directory selection box, because it doesn't seem to be particularly useful: there's no directory tree navigation, instead it's way easier to pass the `--directory` parameter (at least the shell has auto completion).

:shipit:

### CODE REVIEW

**Comment by hammerstefan** [comment]
I think these changes are really useful
1) Drop directory selection dialog - I've never once used the dialog and always smashed enter past it, and I agree that `--directory` is just easier to use
2) Filter by remote - yes, please. This is a great quality of life improvement for when I have many LP MPs open.

I'm pulling your branch locally now to test out the changes myself.

I wonder if this PR wouldn't be easier to review if it was split into 3 parts (remote filter, drop directory dialog, early exit refactor), but that's @philroche 's call if this PR is too complex (doing too much).

**Comment by ikapelyukhin** [comment]
> I wonder if this PR wouldn't be easier to review if it was split into 3 parts (remote filter, drop directory dialog, early exit refactor), but that's @philroche 's call if this PR is too complex (doing too much).

There are two commits in this PR and they can be reviewed separately. The first commit resulted in a large diff due to a change in indentation, but it didn't change anything functionally.

**Comment by hammerstefan** [comment]
I tested the changes locally and everything works great. This PR gets my
:+1: :+1:

File: lpshipit.py [hunk 1]
```diff
@@ -143,153 +157,149 @@ def lpshipit(directory, source_branch, target_branch, mp_owner, debug):
...
+    if not mp_summaries:
+        print("You have no Merge Proposals in either "
+              "'Needs review' or 'Approved' state")
+        sys.exit(1)
+
+    # filter MPs that aren't related to the chosen directory based on the
+    # URLs of git repo remotes
+    repo = git.Repo(directory)
+    remotes = convert_remotes_to_lp_urls(repo)
+    mp_summaries = filter(
```
Comment (j5awry) [inline lpshipit.py:181]: ordering feels weird:

we get MP summaries, check things, filter, do the same check...

I think this should be re-ordered 

1. check directory
2. get proposals
3. filter proposals
4. check for summaries

File: lpshipit.py [hunk 2]
```diff
@@ -143,153 +157,149 @@ def lpshipit(directory, source_branch, target_branch, mp_owner, debug):
...
 
-        def urwid_exit_on_q(key):
-            if key in ('q', 'Q'):
-                raise urwid.ExitMainLoop()
+    if not os.path.isdir(directory):
+        print("'%s' is not a directory" % directory)
+        sys.exit(1)
 
-        def urwid_exit_program(button):
+    if not mp_summaries:
```
[the remark this answers is not in the document]
Comment (ikapelyukhin) [reply lpshipit.py:172]: Yep, dropped.

File: lpshipit.py (hunk 1 above)
Comment (ikapelyukhin) [reply lpshipit.py:181]: I've moved directory-related stuff to the top.

**Review by philroche — APPROVED** [review]
Thanks for this

**Pull request merged**